### ISSUE TITLE
Validation fails if config name does not end in .yaml

### ISSUE THREAD

**Issue by ChristopherRogers1991** [ultralytics/ultralytics#4081]
### Search before asking

- [X] I have searched the YOLOv8 [issues](https://github.com/ultralytics/ultralytics/issues) and found no similar bug report.

### YOLOv8 Component

Validation

### Bug

When using `yolo detect train data=/path/to/config.yaml...`, a validation step is automatically performed after training. This validation step [requires the data file to end in `.yaml`](https://github.com/ultralytics/ultralytics/blob/main/ultralytics/engine/validator.py#L129). The train step does not require this, however, so I initially tried with `config.yml` (yml vs yaml). The training completes, but then I got the error below:

```
# yolo detect train data=/opt/workspace/config.yml model=yolov8n.pt device=0
...
100 epochs completed in 0.029 hours.                                                                                                                                                          
Optimizer stripped from /usr/src/ultralytics/runs/detect/train/weights/last.pt, 6.2MB                                                                                                         
Optimizer stripped from /usr/src/ultralytics/runs/detect/train/weights/best.pt, 6.2MB

Validating /usr/src/ultralytics/runs/detect/train/weights/best.pt...                                                                                                                          
Ultralytics YOLOv8.0.145 🚀 Python-3.10.11 torch-2.0.1 CUDA:0 (NVIDIA GeForce GTX 1650 Ti, 3904MiB)                                                                                           
Model summary (fused): 168 layers, 3006233 parameters, 0 gradients, 8.1 GFLOPs                                                                                                                
Traceback (most recent call last):                                                                                                                                                            
  File "/opt/conda/bin/yolo", line 33, in <module>                                                                                                                                            
    sys.exit(load_entry_point('ultralytics', 'console_scripts', 'yolo')())                                                                                                                    
  File "/usr/src/ultralytics/ultralytics/cfg/__init__.py", line 423, in entrypoint                                                                                                            
    getattr(model, mode)(**overrides)  # default args from model                                                                                                                              
  File "/usr/src/ultralytics/ultralytics/engine/model.py", line 377, in train                                                                                                                 
    self.trainer.train()                                                                                                                                                                      
  File "/usr/src/ultralytics/ultralytics/engine/trainer.py", line 192, in train
    self._do_train(world_size)                                                                 
  File "/usr/src/ultralytics/ultralytics/engine/trainer.py", line 400, in _do_train      
    self.final_eval()                          
  File "/usr/src/ultralytics/ultralytics/engine/trainer.py", line 555, in final_eval
    self.metrics = self.validator(model=f)                                                     
  File "/opt/conda/lib/python3.10/site-packages/torch/utils/_contextlib.py", line 115, in decorate_context
    return func(*args, **kwargs)                                                                                                                                                              
  File "/usr/src/ultralytics/ultralytics/engine/validator.py", line 134, in __call__
    raise FileNotFoundError(emojis(f"Dataset '{self.args.data}' for task={self.args.task} not found ❌"))
FileNotFoundError: Dataset '/opt/workspace/config.yml' for task=detect not found ❌
```

This is especially confusing because the error thrown is `FileNotFoundError`, but the file definitely exists, and was used for the training already.

As far as I can tell, the extension being `.yaml` is undocumented and not really a requirement - just a quirk of how the inference is being handled.

### Environment

**OS:** Ubuntu 23.04
**Nvidia-container-tools:** 1.13.5-1
**Nvidia driver:** 535.54.03-0ubuntu
**Kernel:** 6.2.0-26-generic
**Docker:** 24.0.5, build ced0996
**Container:** ultralytics/ultralytics:latest

### Minimal Reproducible Example

With any otherwise valid config/training setup, change the extension from `yaml` to `yml:

```
yolo detect train data=/path/to/config.yml model=yolov8n.pt
```

### Are you willing to submit a PR?

- [X] Yes I'd like to help by submitting a PR!

**Comment by glenn-jocher** [comment]
@ChristopherRogers1991 thank you for reporting this issue and for your willingness to contribute to the resolution of this issue by submitting a PR. From your description, it seems the validation step of YOLOv8 requires configuration files to be named with `.yaml` extension rather than `.yml`. However, this is not explicitly documented and not a requirement for the training step, which can result in confusion. 

It may indeed seem like a quirk, but often software tools can be very particular about file extensions for various reasons. In this case, the validation step is likely configured to specifically search for `.yaml` files. I understand this can be frustrating, especially when the error message isn't clear.

As for a potential fix, we could look into making the validation step more flexible by accepting both `.yaml` and `.yml` extensions. Additionally, improving the error message to provide more clarity on the issue could be valuable. You're welcome to implement these changes if you'd like. We greatly appreciate your willingness to help improve the project.

**Comment by glenn-jocher** [comment]
@ChristopherRogers1991 great news 😃! Your original issue may now be resolved ✅ in PR #4086. 

To get this update:
- **[Git](https://github.com/ultralytics/ultralytics)** – Run `git pull` from within your `ultralytics/` directory or run `git clone https://github.com/ultralytics/ultralytics` again
- **[Pip](https://pypi.org/project/ultralytics/)** – Update with `pip install -U ultralytics`
- **[Notebooks](https://github.com/ultralytics/ultralytics/blob/main/examples/tutorial.ipynb)** – Check out the updated notebooks  <a href="https://console.paperspace.com/github/ultralytics/ultralytics"><img src="https://assets.paperspace.io/img/gradient-badge.svg" alt="Run on Gradient"></a> <a href="https://colab.research.google.com/github/ultralytics/ultralytics/blob/main/examples/tutorial.ipynb"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"></a> <a href="https://www.kaggle.com/ultralytics/yolov8"><img src="https://kaggle.com/static/images/open-in-kaggle.svg" alt="Open In Kaggle"></a>
- **[Docker](https://hub.docker.com/r/ultralytics/ultralytics)** – Run `sudo docker pull ultralytics/ultralytics:latest` to update your image <a href="https://hub.docker.com/r/ultralytics/ultralytics"><img src="https://img.shields.io/docker/pulls/ultralytics/ultralytics?logo=docker" alt="Docker Pulls"></a>

Thank you for spotting this issue and letting us know. Please confirm if this update fixes the issue for you, and don't hesitate to report any other **issues** you find or **feature requests** you may have. Happy training with YOLOv8 🚀!
